internal/cluster/distribution/eks/ekscluster/nodepools/wait_for_asg.go

```diff
@@ -80,7 +81,21 @@ func WaitForASGToBeFulfilled(
 					return nil
 				}
 			} else {
-				return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s", waitAttempts, waitInterval)
+				asGroup, err := m.GetAutoscalingGroupByStackName(asgName)
+				if err != nil {
+					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get ASG in stack: %s", waitAttempts, waitInterval, asgName)
+				}
+				lastActivity, _ := m.LastAutoscalingActivity(ctx, asGroup.AutoScalingGroupName)
+				if lastActivity != nil {
```

Comment:
**Required**: I would check for error here (and in the next file as well at the same point in the logic), because if we don't have the last activity, we should return the generic error message like above, otherwise proceed with the detailed error in the current if branch.

While we are here it is already an error case (timeout) so I would put the error checking into an if branch and let the detailed error be returned in case the activity is successfully retrieved.

Reply:
You are totally right, sorry for not thinking that through!